### PR TITLE
[Editor] Fix the resizing of an editor when it's rotated (bug 1847268)

### PR DESCRIPTION
**Pull request by calixteman**
There are 2 rotation we've to deal with: the viewer one and the editor one. The previous implementation was a bit complex and having to deal with these rotation would have potentially increase it.
So this patch aims to simplify the implementation and deal with all the possible cases.
The main idea is to transform the mouse deltas according to the rotations and then apply the resizing in the page coordinates system.

### CODE REVIEW

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
It seems that the style changes rely heavily on CSS nesting, a feature that's new enough that I cannot find a dedicated MDN page for it. Looking at https://caniuse.com/css-nesting confirms that browser support isn't great yet, since e.g. even Firefox ESR cannot use this.

That said, we might still be able to use this iff it can be polyfilled automatically. There seems to be a PostCSS package available, however I've not tested if it works, hence please:
 - Add the https://www.npmjs.com/package/postcss-nesting package, see also https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-nesting#readme
 - Import it in the `gulpfile` similar to how other PostCSS plugins are handled.
 - Search for `postcssDirPseudoClass()` and modify *all the relevant* call-sites to include the new plugin; e.g. https://github.com/mozilla/pdf.js/blob/399475247feeffb3184d1c865530930a8638aa0e/gulpfile.mjs#L970-L972 would become
   ```js
     .pipe(
        postcss([
          postcssDirPseudoClass(),
          postcssNesting(),
          autoprefixer(AUTOPREFIXER_CONFIG)
        ])
      )
    ```
 - Build the GENERIC viewer and check that the relevant CSS rules are rewritten as expected.

File: test/integration/stamp_editor_spec.js [hunk 1]
```diff
@@ -37,9 +38,8 @@ describe("Stamp Editor", () => {
       await Promise.all(
         pages.map(async ([browserName, page]) => {
           if (browserName === "firefox") {
-            pending(
-              "Disabled in Firefox, because of https://bugzilla.mozilla.org/1553847."
-            );
+            // Disabled in Firefox, because of https://bugzilla.mozilla.org/1553847.
+            return;
```
Comment (Snuffleupagus) [inline test/integration/stamp_editor_spec.js:42]: Here, and elsewhere below, why is `pending` being replaced with early return?

File: src/display/editor/editor.js [hunk 2]
```diff
@@ -457,10 +461,16 @@ class AnnotationEditor {
...
+    const savedCursor = this.div.style.cursor;
+    this.div.style.cursor = this.parent.div.style.cursor =
+      window.getComputedStyle(event.target).cursor;
+
     const pointerUpCallback = () => {
-      // Stop the undo accumulation in order to have an undo action for each
-      // resize session.
-      this._uiManager.stopUndoAccumulation();
       this._isDraggable = savedDraggable;
       this.parent.div.classList.remove(resizingClassName);
```
Comment (Snuffleupagus) [inline src/display/editor/editor.js:475]: Looking at the updated CSS, do we still use the `resizingClassName` for anything?

If not this line, together with https://github.com/mozilla/pdf.js/blob/915352ec2228be5d836b7a405c633112a0bbe57a/src/display/editor/editor.js#L454-L457, can probably be removed now.

File: web/annotation_editor_layer_builder.css [hunk 3]
```diff
@@ -186,102 +186,226 @@
...
+      &.bottomLeft {
+        cursor: se-resize;
+      }
+
+      &.middleLeft {
+        cursor: s-resize;
+      }
+    }
+  }
 }
```
[the remark this answers is not in the document]
Comment (Snuffleupagus) [reply web/annotation_editor_layer_builder.css:411]: This should be significantly improved, enough to call it fixed I think, when this patch is rebased on PR #16799.

**Comment by calixteman** [comment]
I just updated the integration test about the cursor name.

**Review by Snuffleupagus — APPROVED** [review]
Seems to work as advertised :-)

r=me, thank you!

**Pull request merged**